ashpd-backend-demo/src/screenshot.rs

```diff
@@ -0,0 +1,127 @@
+use ashpd::{
+    backend::{
+        request::RequestImpl,
+        screenshot::{ColorOptions, ScreenshotImpl, ScreenshotOptions},
+    },
+    desktop::{
+        screenshot::{Color, Screenshot as ScreenshotResponse},
+        Response,
+    },
+    AppID, WindowIdentifierType,
+};
+use async_trait::async_trait;
+
+mod shell_screenshot {
+    use zbus::dbus_proxy;
+
+    use super::Color;
+
+    #[dbus_proxy(
+        interface = "org.gnome.Shell.Screenshot",
```

Comment:
Please don't use any private GNOME APIs in components that are not version locked with the rest of GNOME. These APIs are not stable and will only work for components in the same release.

Reply:
This is barely a demo to try out the API, we are not planning to make use of that anywhere, so I think that is fine ?

Reply:
That demo will only work if you run the shell with the "unsafe" mode activated, until it changes the next time, unless you pretend to be a component you're not, but it still raises some warning flags to use these APIs outside the platform. Does a demo really need to use "real" implementations to demonstrate the API, instead of just returning some predefined image?

Reply:
Agree with what you said here. The scope of this PR went too big that it would take forever to ever land. I think we should just focus on providing building blocks of writing portals. Keep the demo as basic as possible.